app/scripts/controllers/transactions/tx-state-manager.js

```diff
@@ -341,6 +341,9 @@ export default class TransactionStateManager extends EventEmitter {
   }
 
   _updateTransactionHistory(txMeta, note) {
+    if (!txMeta.history) {
+      txMeta.history = [];
```

Comment:
Is it cleaner to simply early return if there is no history?

The current logic assumes you already have the initial state entry in the list so I worry this just adds more edge cases and would require us to update the logic of this method.

Reply:
That's right I will remove this but will just check if `history` is defined just to be extra safe for now. I will take into account your comment in core. I agree that flow is a bit weird to update the history at this point.